CHANGELOG.md

```diff
@@ -28,6 +28,55 @@
   reserved word in Erlang/OTP 27.
   ([Jake Barszcz](https://github.com/barszcz))
 
+- Fault tolerance is now at a statement level instead of a function level in most cases. This means that even if there are errors in a single statement, the compiler will attempt to infer the rest of the function. The exceptions to this are if the right hand side of a use is an invalid function or if the left hand side of an assignment is an invalid pattern.
+
+  ```
+  pub fn main() {
+    let a = 1 + 2.0
+    let b = a + 2.0
+    a + b
+  }
+  ```
+
+  Would output:
+
+  ```
+  error: Type mismatch
+    ┌─ /path/src/test_gleam.gleam:2:15
+    │
+  2 │   let a = 1 + 2.0
+    │               ^^^
+
+  The + operator expects arguments of this type:
+
+      Int
+
+  But this argument has this type:
+
+      Float
+
+  Hint: the +. operator can be used with Floats
+
+
+  error: Type mismatch
+    ┌─ /path/src/test_gleam.gleam:3:15
+    │
+  3 │   let b = a + 2.0
+    │               ^^^
+
+  The + operator expects arguments of this type:
+
+      Int
+
+  But this argument has this type:
+
+      Float
+
+  Hint: the +. operator can be used with Floats
+  ```
```

Comment:
I think we can skip the error examples here as there's no new error to display, and the text explains it clearly.